core/src/apps/tezos/sign_tx.py

```diff
@@ -21,14 +21,43 @@
     node = keychain.derive(msg.address_n, CURVE)
 
     if msg.transaction is not None:
-        to = _get_address_from_contract(msg.transaction.destination)
-        await layout.require_confirm_tx(ctx, to, msg.transaction.amount)
-        await layout.require_confirm_fee(
-            ctx, msg.transaction.amount, msg.transaction.fee
-        )
+        # if the tranasction oprtation is used to execute code on a smart contract
+        if msg.transaction.parameters_manager is not None:
+            parameters_manager = msg.transaction.parameters_manager
+
+            # operation to delegate from a smart contract with manager.tz
+            if parameters_manager.set_delegate is not None:
+                delegate = _get_address_by_tag(parameters_manager.set_delegate)
+                await layout.require_confirm_delegation_baker(ctx, delegate)
+                await layout.require_confirm_set_delegate(ctx, msg.transaction.fee)
+
+            # operation to remove delegate from the smart contract with manager.tz
+            if parameters_manager.cancel_delegate is not None:
```

Comment:
does this mean that you can set and cancel delegate and perform a transfer in a single transaction?

Reply:
That was not intended. I'll fix it